ecl/eclcc/eclcc.cpp

```diff
@@ -1498,7 +1500,7 @@ void EclCC::processFile(EclCompileInstance & instance)
     Owned<ISourcePath> sourcePath = optNoSourcePath ? NULL : createSourcePath(curFilename);
     Owned<IFileContents> queryText = createFileContentsFromFile(curFilename, sourcePath, false, NULL);
     const char * queryTxt = queryText->getText();
-    if (optArchive || optGenerateDepend || optSaveQueryArchive)
+    if (optArchive || optGenerateDepend || optSaveQueryArchive || optCheckDirty)
```

Comment:
It does because it uses the archive creation code to do the checking of version.